src/ucm/util/replace.c

```diff
@@ -25,7 +25,20 @@
 #define MAP_FAILED ((void*)-1)
 #endif
 
+#ifdef PTHREAD_RECURSIVE_MUTEX_INITIALIZER_NP
 pthread_mutex_t ucm_reloc_get_orig_lock = PTHREAD_RECURSIVE_MUTEX_INITIALIZER_NP;
+#else
+pthread_mutex_t ucm_reloc_get_orig_lock;
+static void ucm_reloc_get_orig_lock_init(void) __attribute__((constructor));
```

Comment:
BTW, you may use UCS_STATIC_INIT macro to simplify code

Reply:
are we sure this will be called before the lock is used? since some libs may have own ctor which calls malloc/free/...

Reply:
If this lock is used, some code from this library must be already executing, right ?  Then constructors should be ordered before other code.  Anyway, we will see when we reach the stage of actually running something.

Reply:
@kostikbel it's not true for code executing from another constructor. there is no guarantee of constructor order.

Reply:
For us it is important that the mutex is initialized before the hooks are installed, not that arbitrary code in constructors have run.  Just in case, I added the lowest possible ctr priority to the initialization.